data/common.yaml

```diff
@@ -14,3 +14,6 @@ profile_gpu::dcgm::install::dcgm_version: "2.4.6-1"  # 2.4.6-1 is known to work,
 profile_gpu::dcgm::telegraf::dcgm_telegraf_port: 8094
 profile_gpu::dcgm::telegraf::dcgm_telegraf_py_port: 5556
 profile_gpu::dcgm::telegraf::enable: true
+
+profile_gpu::nhc::manage_scripts: true
+profile_gpu::nhc::scripts: {}
```

Comment:
Are the nhc scripts expected to be totally unique across the clusters or are we expecting them to share some scripts?

If there are scripts we think will be the same across clusters I think we should define the script in this module, that way improvements/bug-fixes to the scripts later down the road and easier to deploy across all the clusters using them. Would also be good to define the script(s) as files instead of defining the content in hiera, similar to how the `user_resource_usage.sh` script is defined in this module : https://github.com/ncsa/puppet-profile_monitoring/commit/089320976c9302d0167b0efb8c374ccbbb7dba49

Reply:
I was expecting we might want the ability to introduce custom scripts. On the other hand, I guess there's no harm in deploying scripts that might not get used. If they're not called by the actual nhc configs then they won't do anything.

Does it make sense to have the following 2 params instead?
Boolean manage_default_scripts — if true, install (or possibly even remove) scripts we consider to be appropriate for all clusters; could include the nvidia check that prompted this new class
Hash custom_scripts — rename the current scripts parameter and let people add custom scripts, if needed

Reply:
Yeah I think having some default scripts with the extra ability to add custom ones makes sense.

Reply:
I've made the change to deploy the nvidia check script as a default script through the module as a file, while keeping the ability for custom scripts.

Tested on mg091.

Reply:
Ah I didn't refresh to see your latest message Jake. I've made the change to pull custom_scripts outside the conditional. 
Tested on mg091.

Reply:
Looks good, will mark this request as resolved.